lib/server/connect.js

```diff
@@ -10,14 +19,26 @@ SubHandlers = {
 var uniqueId = 0;
 var serverId = Random.id();
 
-Meteor.publish('kadira.debug.init', function(browserId, clientId) {
+Meteor.publish('kadira.debug.init', function(browserId, clientId, accessToken) {
   check(browserId, String);
   check(clientId, String);
+  check(accessToken, Match.Any);
 
   var kadiraInfo = Kadira._getInfo();
   if(kadiraInfo) {
     var sessionId = kadiraInfo.session;
 
+    if(AppConfig.env === 'production') {
+      var tokenKey = AppConfig.authKey + "_" + accessToken;
+      if(AppConfig.accessToken.get(tokenKey) !== accessToken) {
+        // If the subcription request with invalid accessToken
+        // return without initiate subscription
+        return;
+      }
+
+      AppConfig._authorizedSessions.set(sessionId, true); // ::
```

Comment:
Don't use another LRU cache for this. Mark the session as a authorized one. Like this:

```
kadiraInfo.sessionAuthorized = true;
```